src/servers/http_server_v2.cc

```diff
@@ -653,24 +648,33 @@ void
 WriteDataToJsonHelper(
     rapidjson::Value* response_output_val,
     rapidjson::Document::AllocatorType& allocator,
-    const rapidjson::Value& shape, int shape_index, T* base, int* counter)
+    const rapidjson::Value& shape, int shape_index, T* base, int* counter,
+    const DataType dtype)
 {
   for (int i = 0; i < shape[shape_index].GetInt(); i++) {
     if ((shape_index + 1) != (int)shape.Size()) {
       rapidjson::Value response_output_array(rapidjson::kArrayType);
       WriteDataToJsonHelper(
           &response_output_array, allocator, shape, shape_index + 1, base,
-          counter);
+          counter, dtype);
       response_output_val->PushBack(response_output_array, allocator);
     } else {
-      rapidjson::Value data_val((T)(base[*counter]));
-      response_output_val->PushBack(data_val, allocator);
-      *counter += 1;
+      if (dtype != TYPE_STRING) {
+        rapidjson::Value data_val((T)(base[*counter]));
+        response_output_val->PushBack(data_val, allocator);
+        *counter += 1;
+      } else {
+        uint32_t* len = reinterpret_cast<uint32_t*>(base + *counter);
+        char* cstr =
+            reinterpret_cast<char*>(base + *counter + sizeof(uint32_t));
```

Comment:
So the returning data will not contain the length bytes?

Reply:
The returning data will be an array (nested) of strings without their length bytes.